README
<insert crazy ascii image + legal shit up here, yay boring>


In order to install rash:
1. Upload all .php and .html files to your webserver. You'll probably want to denote it to its own directory.
2. Run install.html inside the directory where you've uploaded the files through a web browser of your choice.
3. Read the instructions of what tables.php will do and how you need to edit config.php, and follow the link.
4. Delete install.html and tables.php from your server for security reasons.
5. Change administrator password.
6. Insert quotes.

CHANGELOG
---From Rash 1.0:
12/06 - p00p
Changed admin login error message to "Login failed."
Replaced the calling of the template.php file with a variable, $templatefile, which is set in config.php.  This consisted of replacing the include to the template in the main file with one to the config file, adding a variable to the config file that specifies the template file, and removing the include to the config file that is in the template file.  I also removed every global variable in index.php.
Streamlined some structural elements of config.php.  Changed a few of the explanations to be more explanatory. :)
Moved $noservercnx, $nodbcnx, and $queryerror to index.php
Moved style variables from config.php to style.php, a part of the optional default template

12/07 - tommah
Fixed Ben's mistake of erasing all the globals in each parent function (function such as home, browse, bla) by including config.php in each function. It was broken because there were no variables defined in the function, causing nothing to happen.

12/07 - p00p
Added a variable $rashversion to the top of index.php.
template.php now shows the version in the bottom bottomtdb.

12/07&&12/08 - tommah
Deleted a variable $bottomjimmy from style.php, and the css class relating to it.
Moved $title from config.php to template.php. It's template related.
Moved all error outputs to config.php, they would need to be globalized in all functions if they were to be used in index.php. Putting them in config.php does that automatically for us.
Added a declaration to style.php about hardcoded css classes.
Added a variable $loginfail to config.php. Error that you get for login failures, obviously, also put it in some places where login errors can occur, not quite sure I got them all.
Added a variable $sessionhandle to config.php. Error that you get for session used for ratingplus, ratingminus, and add quote.
Added a variable $rateplusyes to config.php to handle the confirm message for upping a rating on a quote.
Added a variable $rateminusyes to config.php to handle the confirm message for downing a rating on a quote.
Changed the color of $rashversion in template.php to a variable $versioncolor stored in style.php and called by a class setting with a span tag in template.php.
Added a variable $newslimit to config.php to set the limit of the number of news posts that the front page displays.

/*
Made a new section in config.php called Default Rash Template Customization.
Added a variable $welcomecolumn to section 3 of config.php to turn the left welcome column on or off. Best I could do was hardcode this :(.
Added a variable $welcomecolumncontent to section 3 of config.php to edit the content of the left column. Best I could do was hardcode this :(.
Added a variable $newscolumn to section 3 of config.php to edit the existance of the news column. Best I could do was hardcode this :(.

We need to come up with a way to get rid of all this shit.
*/
I renamed install.html and tables.php to install.what.html and install.what.php to mask them from the public. If tables is run, all our shit is erased.

12/10 - tommah
Began extracting all HTML from index.php and puting in a file called output.php, this file will control the html outputs generated by the sql queries, include functions that will be called in each section function, and operate at what I hope will be smoothly.
Added $outputfile to config.php to declare what the output file is.
^^^^^^^ <-- working on it

12/13 - p00p
Added two variables:
$versionnumber contains the version number (1.0 in a 1.0-devel version)
$subversion is the devel in a 1.0 release
both those two and $rashversion in index.php must be edited for a new release

12/14 - p00p
Changed the legal message on all the php files it was in, and added it to 
a couple that it wasn't in.  (It should be in all of them now.)

12/15 - tommah
Made a templates folder. All template files should go there, since it's  just easier to manage.
Began changing <<ALL>> the ways quotes can be seen throughout index.php to rely on output.php to execute the code necessary to view them. This should lower filesize immensely.

12/23 - tommah
Moved adminpanel to respective templates rash_admin and bash_admin. (they're the same right now :-/)
Made the adminpanel nicer looking. (Both)
Made rash_template and bash_template their own respective folders in /templates.
Removed that $title variable from *_template.php.
Removed/renamed a lot of instances of stupid CSS classes in *_style.php and index/output.php so they make sense or are not redundant.
Set up all the $section variables so they can be called in templates with a switch in *_template.php.

12/28 - tommah
Added variable $flag_confirm to config.php
Added variable $cookie_time to config.php
Added variable $add_user_confirm to config.php
Added variable $not_loggedin_in to config.php
Added variable $add_news_confirm to config.php
Added variable $password_mismatch to config.php
Added variable $approve_quote_confirm to config.php
Added variable $quote_deleted to config.php
Added variable $deflag_confirm to config.php
Changed $sessionhandle to $cookie_already_set
Changed $quoteyes/no or whatever to $upgrade/downgrade_confirm
Made CREDITS file.
I changed every confirmation/error output, changed it into a variable, and put it in config.php so it can be changed easily by editing config.php.
Made all SQL queries in index.php awesome, relying on output from phpmyadmin for a template on query structure rather than my old makeshift ones.
Added new function qotw. Displays all quotes submitted within the last week.
Added new column to rash_quotes table in the mysql databse: date. It stores a unix timestamp to compare for the qotw function and could be called to show dates in quotes and such.
Split config.php into two more parts: part 3 Errors part 4 confirmation messages.
Got rid of $misuse from config.php, it wasn't used.

12/29 - tommah

output.php
<?php
/* 
RQMS - Rash Quote Management System
Copyright (C) 2003-2004 Tom Cuchta (tommah@instable.net / http://www.mastergoat.com) and Instable Network (p00p@instable.net / http://www.instable.net)

http://rqms.sourceforge.net

This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.

You should have received a copy of the GNU General Public License along with this program; if not, write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
*/

############
##FRONT PAGE
############

function front_page_output(){		// Usually the first page you see. It's configured for use as Bash.org and the default rash
	global $section;				// And Bash templates. Change at your own will.
	$welcomecolumn = '1';			// Turning this to anything but 1 turns the welcome column off the main page of your site. The news updates will stretch across the empty space.
	$newscolumn = '1';				// Turning this to anything but 1 turns the news updates off and the welcome page will stretch across the empty space.
	include('config.php');
	if($welcomecolumn == '1'){
?>
		<div id="news_format">
			<div id="frontpage_left">
				<p><span style="font-weight:bold">Hello!</span> Enjoy our collection of quotes and if you feel the pleasure, you can add some of your own. 	</p>
				<p>Depending on if you like or dislike a quote, there are links that are + and - signs that you can click to promote or demote a quote. If you think a quote has slipped by our highly technological sensors and isn't appropriate for this website, you can always click the [X] link near it. <br /> Note: The quote is not removed from the database, it just puts a flag up for the admins to check it to verify that it is alright, it may or may not be removed. </p>
				Queries? Remarks? <a href="mailto:<?=$email?>"><?=$email?></a>
			</div>
			<div id="frontpage_right">
<?php
	}
	if($newscolumn){
	$link = mysql_connect("$hostname", "$username", "$dbpasswd")
			or die($noservercnx . mysql_error());
		if (! @mysql_select_db("$dbname") ) {
			echo($nodbcnx);
			exit();
		}
		$sql = "SELECT * FROM `" . $newstable . "` ORDER BY `id` desc LIMIT " . $newslimit;
		$result = database_connect($sql);
		while($row = mysql_fetch_array($result)){
			echo nl2br("<p class=\"news_date\">" . $row["date"] . "</p>\n"); 
			if(login_check(0)){
				echo nl2br(	"<a href=\"?news_edit" . $GET_SEPARATOR_HTML . "id=" . $row['id'] . "\">[Edit]</a>&nbsp;" .
							"<a href=\"?news_delete" . $GET_SEPARATOR_HTML . "id=" . $row['id'] . "\">[Delete]</a>");
			}
			echo nl2br("<div class=\"news_post_format\">" . $row["news"] . "</div>\n");
		}
		echo "</div>\n</div>\n";
	}
}



#############
##ADMIN LOGIN
#############

function preadmin_output(){								//	Displayed at ./?admin IF the user is not logged in		Query for admin/mod login.
print <<<EOF
       Welcome Guest!<br />								
       If your account says "does not exist" or something of the sort, try using all lowercase letters. The usernames ARE case sensitive.<br />
       <form method="post" action="?adminlogin">
        User Name: <input type="text" name="user" size="28" class="basicinput"><br />
        <p>
         Password:&nbsp;&nbsp; <input type="password" name="password" size="28" class="basicinput"><br />
        </p>
        <input type="submit" name="submit" class="basicsubmit">
       </form>
EOF;
}

################
##ADD QUOTE PAGE
################

function add_quote_page_output(){						//	Displayed at ./?add		Query for users to add quotes
print <<<EOF
       Please remove timestamps unless necessary.
      </div>
     </td>
    </tr>
    <tr>
     <td>
      <div align="left">
       <form method="post" action="?added">
        <textarea cols="80" rows="5" name="quote" class="basicinput"></textarea><br />
        <input type="submit" value="Add Quote" class="basicsubmit">
        <input type="reset" value="Reset" class="basicsubmit"><br />
       </form>
EOF;
}

####################
##SEARCH QUOTES PAGE
####################

function search_quotes_page_ouput(){					//	Displayed at ./?search		The search query for the user
print <<<EOF
       <form method="post" action="?searched">
        <input type="text" name="search" size="28" class="basicinput">
        <input type="submit" name="submit" class="basicsubmit"><br />
        Sort the quotes by: <select name="sortby" size="1">
         <option selected>rating
         <option>id
        </select>&nbsp;&nbsp;&nbsp;&nbsp;
		How many quotes to show?: <select name="number" size="1">
         <option selected>10
         <option>25
         <option>50
         <option>75
         <option>100
        </select>
       </form>
EOF;
}


########################
##ADD QUOTE CONFIRMATION
########################

function add_quote_confirmation($qQUOTE){
	include('config.php');
print <<<EOF
       Thanks, your quote has been submitted.<br />
       For verification, the quote you have submitted is<br />
EOF;
	echo nl2br(
		"<pre>" . $qQUOTE . "</pre>\n"
	);
print <<<EOF
       if this is not the quote you have submitted or planned to submit, send an e-mail to <a href="mailto:$email">$email</a>, and explain your problem.<br />
       Thank you.
EOF;
}

##############
##QUOTE FORMAT
##############

function quote_format($sql, $qotw_check){								//	Displayed anywhere quotes are returned.
	include('config.php');
	global $qID;
	$result = database_connect($sql);
	while ( $row = mysql_fetch_array($result) ) {
		echo(
			"       <a href=\"?" . $row['id'] . "\">#" . $row['id'] . "</a>\n" .
			"       <a href=\"?ratingplus" . $GET_SEPARATOR_HTML . "id=" . $row['id'] . "\" style=\"text-decoration:none;\">+ </a>\n" . 
			"       (" . $row['rating'] . ")\n" .
			"       <a href=\"?ratingminus" . $GET_SEPARATOR_HTML . "id=" . $row['id'] . "\" style=\"text-decoration:none;\">-</a>\n" .
			"       <a href=\"?flag" . $GET_SEPARATOR_HTML . "id=" . $row['id'] . "\" style=\"text-decoration:none;\">[X]</a>\n"
		);
		if($qotw_check){
			echo "       <span style=\"font-weight:bold\">Submitted on " . date ("M d Y", $row['date']) . " at " . date ("g:i A", $row['date']) . "</span>";
		}
		if(login_check(0)){
			echo(
				"	<a href=\"?adelete" . $GET_SEPARATOR_HTML . "id=" . $row['id'] . "\" style=\"text-decoration:none\">[Delete]</a>\n" .
				"	<a href=\"?aedit" . $GET_SEPARATOR_HTML . "id=" . $row['id'] . "\" style=\"text-decoration:none\">[Edit]</a>\n"
			);
		}
		echo nl2br(
			"<br /><div class=\"quote_output\">" . $row["quote"] . "</div>"
		);
	}
}

############
##QUOTE_EDIT
############
function quote_edit($sql){
	include('config.php');
	while ( $row = mysql_fetch_array($result)){
print <<<EOF

      <form action="?aedit" method="post">
       <textarea cols="70" rows="15" name="quote" class="basicinput">
EOF;
echo $row['quote'];
print <<<EOF
</textarea><br />
       Edit the quote.<br /><br />
       <input type="text" size="4" name="rating" class="basicinput" value="
EOF;
echo $row['rating'];
print <<<EOF
"><br />
       Edit the quote's rating.<br /><br />
       <input type="submit" class="basicsubmit" value="Submit">
       <input type="hidden" value="1" name="editresult">
       <input type="hidden" value="
EOF;
echo $row['id'];
print <<<EOF
" name="id">
      </form>
EOF;
	}
}

############
##NEWS_CHANGE
############
function news_change($sql){
	include('config.php');
	$link = mysql_connect("$hostname", "$username", "$dbpasswd")
		or die($noservercnx . mysql_error());
	if (! @mysql_select_db("$dbname") ) {
		echo $nodbcnx;
		exit();
	}
	$result = mysql_query($sql)
		or die($queryerror . mysql_error());
	while($row = mysql_fetch_array($result)){
print <<<EOF
      <form action="?news_edit" method="post">
	   <textarea cols="70" rows="15" name="news" class="basicinput">
EOF;
echo $row['news'];
print <<<EOF
</textarea><br />
      Edit the news.<br /><br />
	  <input type="text" size="4" name="date" class="basicinput" value="
EOF;
echo $row['date'];
print <<<EOF
"><br />
		Edit the news' date.<br /><br />
		<input type="submit" class="basicsubmit" value="Submit">
		<input type="hidden" value="1" name="editresult">
		<input type="hidden" value="
EOF;
echo $row['id'];
print <<<EOF
" name="id">
	</form>
EOF;
	}
}
?>
